### PR TITLE
refactor: improve transaction Done call API

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Improve transaction `Done` call API.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3720)

### CODE REVIEW

File: pkg/storagev2/leveldbstore/transaction.go [hunk 1]
```diff
@@ -106,19 +106,14 @@ func (s *TxStore) Delete(item storage.Item) error {
...
 
 // Rollback implements the Tx interface.
 func (s *TxStore) Rollback() error {
-	if err := s.IsDone(); err != nil {
+	if err := s.TxState.Done(); err != nil {
 		return err
 	}
-	defer s.TxState.Done()
 
 	s.opsMu.Lock()
```
Comment (aloknerurkar) [inline pkg/storagev2/leveldbstore/transaction.go:118]: Do we need this mutex lock anymore?

File: pkg/storagev2/inmemchunkstore/transaction.go [hunk 2]
```diff
@@ -81,19 +81,14 @@ func (s *TxChunkStore) Delete(ctx context.Context, addr swarm.Address) error {
...
 
 // Rollback implements the Tx interface.
 func (s *TxChunkStore) Rollback() error {
-	if err := s.IsDone(); err != nil {
+	if err := s.TxState.Done(); err != nil {
 		return err
 	}
-	defer s.TxState.Done()
 
 	s.opsMu.Lock()
```
Comment (aloknerurkar) [inline pkg/storagev2/inmemchunkstore/transaction.go:93]: Same, I dont think we need this mutex Lock anymore right?

**Review by aloknerurkar — APPROVED** [review]

File: pkg/storagev2/leveldbstore/transaction.go (hunk 1 above)
Comment (mrekucci) [reply pkg/storagev2/leveldbstore/transaction.go]: Good catch! Fixed.

File: pkg/storagev2/inmemchunkstore/transaction.go (hunk 2 above)
Comment (mrekucci) [reply pkg/storagev2/inmemchunkstore/transaction.go]: Fixed.

**Pull request merged**